### PR TITLE
Bluetooth: Mesh: fix SRPL clearing

### PR DESCRIPTION
**Pull request by KKopyscinski**
`item_clear` should clear adresses from primary to primary + len. Change lt to le comparison in for loop.

### CODE REVIEW

File: subsys/bluetooth/mesh/sol_pdu_rpl_srv.c [hunk 1]
```diff
@@ -67,8 +67,12 @@ static int item_clear(struct bt_mesh_model *mod,
 		return -EINVAL;
 	}
 
-	for (int i = 0; i < len; i++) {
-		bt_mesh_srpl_entry_clear(primary + i);
+	if (len == 0) {
+		bt_mesh_srpl_entry_clear(primary);
+	} else {
+		for (int i = 0; i < len; i++) {
```
Comment (alxelax) [inline subsys/bluetooth/mesh/sol_pdu_rpl_srv.c:73]: I guess this will be sufficient
```suggestion
		for (int i = 0; i < len || len == 0; i++) {
```

Comment (KKopyscinski) [reply subsys/bluetooth/mesh/sol_pdu_rpl_srv.c:73]: Yeah, I thought that something like this misses some edge case but now looks ok to me. But isn't this just same as `for (int i = 0; i <= len; i++) {`?

Comment (alxelax) [reply subsys/bluetooth/mesh/sol_pdu_rpl_srv.c:73]: > `for (int i = 0; i <= len; i++) {`?
This is different, upper boundary shouldn't be included

Comment (KKopyscinski) [reply subsys/bluetooth/mesh/sol_pdu_rpl_srv.c:73]: Oh, right, this was this edge case :)

Comment (alxelax) [reply subsys/bluetooth/mesh/sol_pdu_rpl_srv.c:73]: Sorry, I made mistake. It should be: 

`for (int i = 0; i < len || i == 0; i++) {`

**Review by alxelax — APPROVED** [review]

**Review by PavelVPV — APPROVED** [review]

**Pull request merged**